server/plugins/queues/shareScrollQueue.ts

```diff
@@ -1,100 +1,97 @@
-import { Worker as BullWorker } from 'bullmq';
+import { Worker as BullWorker, type Job } from 'bullmq';
 import { watch } from 'fs';
 import { shareScrollQueue } from '~/server/queue';
-import { Worker } from 'worker_threads';
+// import { Worker } from 'worker_threads';
 import { db } from '~/server/db';
 import { bannerJobsTable } from '~/database/schema';
 import type {
   WorkerInput,
-  WorkerFinished,
-  WorkerResponse,
-  WorkerError,
+  // WorkerFinished,
+  // WorkerResponse,
+  // WorkerError,
+  // since i'm not using these, should they be removed from the types file too?
 } from '~/workers/shareScrollWorkerTypes';
 
 const {
   redis: { host, port },
-  clientSecret,
 } = useRuntimeConfig();
 
 const round = (num: number | null) =>
   num !== null ? parseInt(num.toFixed(0)) : null;
 
-function isJobFinished(message: WorkerResponse): message is WorkerFinished {
-  return message.type === 'jobFinished';
-}
+// function isJobFinished(message: WorkerResponse): message is WorkerFinished {
+//   return message.type === 'jobFinished';
+// }
 
-function isJobError(message: WorkerResponse): message is WorkerError {
-  return message.type === 'error';
-}
+// function isJobError(message: WorkerResponse): message is WorkerError {
+//   return message.type === 'error';
+// }
+
+// i ended up not using these but should i?
 
 export default defineNitroPlugin(async () => {
   const createWorker = () =>
-    new Worker('./workers/shareScrollWorker.js', {
-      workerData: { username: null, filePath: null },
-    });
+    new BullWorker<WorkerInput>(
+      'shareScrollQueue',
+      '/src/workers/shareScrollWorker.js',
+      {
+        useWorkerThreads: true,
+        connection: {
+          host,
+          port: Number(port),
+        },
+      }
+    );
 
   let shareScrollWorker = createWorker();
 
   if (import.meta.dev) {
-    watch('./workers/shareScrollWorker.js', () => {
-      shareScrollWorker.terminate();
+    watch('/src/workers/shareScrollWorker.js', () => {
```

Comment:
was there a reason this was changed?

Reply:
Needed an absolute path because a relative path had the code looking for `workers/` somewhere very different.
```
ERROR  Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/src/node_modules/bullmq/dist/cjs/classes/workers/shareScrollWorker.js' imported from /src/node_modules/bullmq/dist/cjs/classes/child-processor.js
hatchery-1  | Did you mean to import "../../../../../workers/shareScrollWorker.js"?
```

Reply:
Oh and if you meant `terminate()` versus `close()`, I'm not sure actually - I was just referencing off the code you provided over Discord and it works. Do we need `terminate()`?